privacyidea/lib/crypto.py

```diff
@@ -165,24 +168,29 @@ def hash(val, seed, algo=None):
 
 
 @log_with(log, log_entry=False, log_exit=False)
-def hash2(password):
+def pass_hash(password):
     """
-    Hash values with argon2
-    :param val:
-    :return:
+    Hash password with crypt context
+    :param password: The password to hash
+    :type password: str
+    :return: The hash string of the password
     """
-    pw_dig = argon2.using(rounds=ROUNDS).hash(password)
+    pw_dig = pass_ctx.hash(password)
```

Comment:
How would we **assure** that argon2 is used!?

What happens if argon2_cffi is not installed. It will fallback to pbkdf2? 
I would rather like to enforce argon2. Can we do this?
With a cryptcontext, that only contains argon2?

Maybe we can have two contexts:

~~~~python
verify_ctx = CryptContext(['argon2', 'pbkdf2_sha512'], argon2__rounds=ROUNDS)
hash_ctx =  CryptContext(['argon2'], argon2__rounds=ROUNDS)
~~~~

To assure hashing with argon2

Reply:
According to the [passlib docs](https://passlib.readthedocs.io/en/stable/narr/context-tutorial.html#basic-usage), it always picks the first entry in the list for hashing unless told otherwise.
If no argon2 backend is installed, hashing fails completely.